packages/web3-eth-accounts/src/index.js

```diff
@@ -264,18 +271,18 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
 
     // Otherwise, get the missing info from the Ethereum Node
     return Promise.all([
-        isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId,
+        ((isNot(tx.common) || isNot(tx.common.customChain.chainId)) ? ( isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId) : undefined ), //tx.common.customChain.chainId is not optional inside tx.common if tx.common is provided
```

Comment:
This is confusing, but we are limited by possibly having to do a async call (`getChainId`), so I guess it's okay for `1.x`

Reply:
```suggestion
        isNot(tx.common) || isNot(tx.common.customChain.chainId) ?
            isNot(tx.chainId) ? _this._ethereumCall.getChainId() : tx.chainId
            : undefined, //tx.common.customChain.chainId is not optional inside tx.common if tx.common is provided
```

I think this formatting makes it a bit more readable